modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py

```diff
@@ -2670,6 +2673,7 @@ def to_pandas(self):
         # restrictions on column names.
         df.columns = self.columns
 
+        self._pandas_df = df
```

Comment:
What about in-place operations executed on returned dataframe? Wouldn't such operations affect stored object?

Reply:
Good catch! You are right, the stored object will be affected.
A non-deep copy should be returned here. It will share the data with the main frame, but will not change it.

Reply:
even after the copy, in-place operations still mutate unwanted frames:
```python
import modin.pandas as pd

def setitem(df, i, val):
    df.iloc[i, 0] = val
    return df

df = pd.DataFrame({"a": [1, 2, 3, 4, 5]})

res1 = df._default_to_pandas(lambda df: setitem(df, 0, 10))
res2 = df._default_to_pandas(lambda df: setitem(df, 1, 100))

print(df)
#      a
# 0   10
# 1  100
# 2    3
# 3    4
# 4    5

print(res1)
#     a
# 0  10
# 1   2
# 2   3
# 3   4
# 4   5

print(res2)
#      a
# 0   10
# 1  100
# 2    3
# 3    4
# 4    5
```
Does it mean that we should copy `pandas_df` anytime someone requests that field or restrict in-place operations? (i'm mostly for the second option)

Reply:
You are right, we need to restrict the inplace operations or create a deep copy in case of an inplace operation.